cdap-app-fabric/src/main/java/io/cdap/cdap/internal/profile/ProfileService.java

```diff
@@ -142,10 +147,13 @@ public Profile getProfile(ProfileId profileId, Map<String, String> overrides) th
    *
    * @param profileId the id of the profile to save
    * @param profile the information of the profile
-   * @throws MethodNotAllowedException if trying to update the Native profile
+   * @throws MethodNotAllowedException if trying to update the Native profile or creating new profile when disallowed.
    */
   public void saveProfile(ProfileId profileId, Profile profile) throws MethodNotAllowedException {
     TransactionRunners.run(transactionRunner, context -> {
+      if (!cConf.getBoolean(Constants.Profile.CREATION_ALLOWED, true)) {
```

Comment:
We usually put defaults in cdap-defaults.xml, after which the default value here doesn't need to be given.

Reply:
also, this method handles but creates and updates. If we only want to block creation it needs to be handled differently. If we want to block creation and updates, then we should change the configuration property name and the error message.

I would think that we want to block updates as well, otherwise users can just change the existing profile.

Reply:
Done.